src/modeler/AutoRest.Swagger/Validation/PropertiesNamesCamelCase.cs

```diff
@@ -28,25 +30,24 @@ public class PropertiesNamesCamelCase : TypedRule<Dictionary<string, Operation>>
         /// Violation category of the Rule.
         /// </summary>
         public override ValidationCategory ValidationCategory => ValidationCategory.RPCViolation;
+
         /// <summary>;
         /// The template message for this Rule. 
         /// </summary>
         /// <remarks>
         /// This may contain placeholders '{0}' for parameterized messages.
         /// </remarks>
-        public override string MessageTemplate => Resources.PropertyNameShouldBeCamelCase;
+        public override string MessageTemplate => Resources.PropertyNameCamelCase;
 
         /// <summary>
         /// The severity of this message (ie, debug/info/warning/error/fatal, etc)
         /// </summary>
         public override Category Severity => Category.Error;
 
-        /// <summary>
-        /// Validates whether property names are camelCase.
-        /// </summary>
-        /// <param name="path">Operation path to validate</param>
-        /// <returns>true if there are no propeties violating camelCase style, false otherwise.</returns>
-        public override bool IsValid(Dictionary<string, Operation> path, RuleContext context, out object[] formatParameters)
+        ///// <summary>
+        ///// Validates whether property names are camelCase.
+        ///// </summary>
+        public override IEnumerable<ValidationMessage> GetValidationMessages(Dictionary<string, Operation> path, RuleContext context)
```

Comment:
So, when you apply this validation rule to the paths only, it will check model properties that are being referenced in appropriate paths; what about nested references? There can also be cases where service teams like to declare forward-looking models that are not being referenced yet, we should be flagging those properties. I think this rule should be applied to the definitions dictionary instead of paths